src/main/clojure/clara/tools/tracing.cljc

```diff
@@ -114,3 +114,48 @@
                          (first))]
     (.-trace ^PersistentTracingListener listener)
     (throw (ex-info "No tracing listener attached to session." {:session session}))))
+
+(defn ^:private node-id->productions
```

Comment:
Similar to my comment for `traced-session->ranked-productions`,
I'd prefer `node-productions` since this doesn't seem like a "conversion" operation.

It's subjective of course.

Reply:
Perhaps "productions-for-id"?